buku

```diff
@@ -200,208 +200,171 @@ class BukuCrypt:
 
+    def __init__(self, iterations=8, dbfile=None, encfile=None, password=None, replace=True):
         try:
             from getpass import getpass
             from hashlib import sha256
 
             from cryptography.hazmat.backends import default_backend
             from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
-        except ImportError:
-            LOGERR('cryptography lib(s) missing')
-            sys.exit(1)
+        except ImportError as e:
+            raise RuntimeError('cryptography lib(s) missing') from e
+        self._sha256, self._default_backend = sha256, default_backend
+        self._Cipher, self._algorithms, self._modes = Cipher, algorithms, modes
+        self._getpass = (getpass if sys.stdin.isatty() else (lambda: sys.stdin.readline().rstrip('\n')))
 
         if iterations < 1:
-            LOGERR('Iterations must be >= 1')
-            sys.exit(1)
-
-        if not dbfile:
-            dbfile = os.path.join(BukuDb.get_default_dbdir(), 'bookmarks.db')
-        encfile = dbfile + '.enc'
-
-        db_exists = os.path.exists(dbfile)
-        enc_exists = os.path.exists(encfile)
-
-        if db_exists and not enc_exists:
-            pass
-        elif not db_exists:
-            LOGERR('%s missing. Already encrypted?', dbfile)
-            sys.exit(1)
-        else:
-            # db_exists and enc_exists
-            LOGERR('Both encrypted and flat DB files exist!')
-            sys.exit(1)
-
-        password = getpass()
-        passconfirm = getpass()
-        if not password or not passconfirm:
-            LOGERR('Empty password')
-            sys.exit(1)
-        if password != passconfirm:
-            LOGERR('Passwords do not match')
-            sys.exit(1)
+            raise RuntimeError('Iterations must be >= 1')
+
+        self.iterations, self.password, self.replace = iterations, password, replace
+        self.dbfile = dbfile or os.path.join(BukuDb.get_default_dbdir(), 'bookmarks.db')
+        self.encfile = encfile or (dbfile + '.enc')
+
+        self._db_exists = os.path.exists(self.dbfile)
+        self._enc_exists = os.path.exists(self.encfile)
+        if self._db_exists and self._enc_exists:
+            raise RuntimeError('Both encrypted and flat DB files exist!')
+
+    def _encrypt_file(self):
+        if not self._db_exists:
+            raise RuntimeError(f'{self.dbfile} missing. Already encrypted?')
+
+        if not self.password:
+            self.password = self._getpass()
+            if not self.password:
+                raise RuntimeError('Empty password')
+            passconfirm = self._getpass()
+            if not passconfirm:
+                raise RuntimeError('Empty password')
+            if self.password != passconfirm:
+                raise RuntimeError('Passwords do not match')
 
         try:
-            # Get SHA256 hash of DB file
-            dbhash = BukuCrypt.get_filehash(dbfile)
+            self._encrypt()
```

Comment:
Also extracted the actual encryption/decryption logic into dedicated methods, to separate them from the wrapping logic